tests/ExcelProvider.Tests/ExcelProvider.Tests.fs

```diff
@@ -380,3 +383,9 @@ let ``HashHeader false with header removed``() =
     let expectedTime = new DateTime(1899, 12, 31, 8, 0, 0)
     row.Column5 |> should equal expectedTime
     row.Column6 |> should equal 100.0M
+
+[<Test>]
+let ``Reading all rows in a well-formed excel file does not return empty rows``() =
+    let file = DifferentMainSheet()
+    // For reference, DifferentMainSheet has 1 header row and 6 data rows.
+    file.Data |> Seq.length |> should equal 6
```

Comment:
This is a pretty simple test. I added it alone at commit <update comment with commit number>, and it was failing in that case. By the end, it is passing, along with the other tests.

As mentioned above, it might be more appropriate to test the length of an Excel file selecting multiple ranges here. If that makes more sense, I believe this would do the trick.

```suggestion
[<Test>]
let ``Reading all rows in a well-formed excel file does not return empty rows``() =
    let fileWithUnboundRange = DifferentMainSheet()
    // For reference, DifferentMainSheet has 1 header row and 6 data rows.
    fileWithUnboundRange.Data |> Seq.length |> should equal 6
    
    let fileWithBoundRange = MultipleRegions()
    fileWithBoundRange |> Seq.length |> should equal 4
```

Reply:
Thank you very much for this extremely detailed and nicely structured pull request.  I am currently a bit time constrained. I have booked out time starting on 11th of September to move this forward. Apologies for the wait.

Reply:
Sounds good! I realize this is probably way over-annotated for what was effectively a 3 line fix, but since all three locations end up modifying the same value I wanted to track the changes through the code.

That sounds good on the timing; no imminent need for this change here. Honestly, I was just using this library and saw an opportunity to dip my toes into open-source contribution on a small enough piece where I felt I could contribute :).